packages/perseus-editor/src/diffs/answer-area-diff.tsx

```diff
@@ -7,15 +7,17 @@ import performDiff from "./shared/widget-diff-performer";
 import type {PerseusAnswerArea} from "@khanacademy/perseus-core";
 
 type AnswerAreaDiffProps = {
-    after: PerseusAnswerArea;
-    before: PerseusAnswerArea;
+    after: PerseusAnswerArea | undefined;
+    before: PerseusAnswerArea | undefined;
     title: string;
 };
 
 export class AnswerAreaDiff extends React.Component<AnswerAreaDiffProps> {
     render(): React.ReactNode {
         const {after, before, title} = this.props;
-        const diff = performDiff(before, after);
+
+        // If before or after is undefined, pass an empty object to performDiff.
+        const diff = performDiff(before ? before : {}, after ? after : {});
```

Comment:
Should we pass undefined directly instead?  Looking at `performDiff`, it already handles undefined directly, which would preserve the `ADDED/REMOVED` semantics rather than treating it as `CHANGED`.
Then it could be simplified to
```suggestion
        const diff = performDiff(before, after);
```

Reply:
oh ok that makes sense, thanks Cat